script.js

```diff
@@ -1,21 +1,18 @@
-//You can edit ALL of the code here
-function setup() {
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
-}
+const state = {
+  episodes: getAllEpisodes(),
+  searchTerm: "",
+  selectedEpisodeId: "",
+};
 
-function makePageForEpisodes(episodeList) {
-  const rootElem = document.getElementById("root");
-  for (const episode of episodeList) {
-    const card = document.getElementById("show-card").content.cloneNode(true);
-    const title = card.querySelector("#show-title");
-    title.textContent = episode.name;
-    title.setAttribute("href", episode.url);
-    card.querySelector("#episode-code").textContent = episodeCode(episode.season, episode.number);
-    card.querySelector("#episode-img").src = episode.image.medium;
-    card.querySelector("#episode-summary").innerHTML = episode.summary;
-    rootElem.append(card);
-  }
+function createEpisodeCard(episode) {
+  const card = document.getElementById("episode-card").content.cloneNode(true);
+  const title = card.querySelector("#episode-title");
+  title.textContent = episode.name;
+  title.setAttribute("href", episode.url);
+  card.querySelector("#episode-code").textContent = episodeCode(episode.season, episode.number);
+  card.querySelector("#episode-img").src = episode.image.medium;
```

Comment:
What will happen if the image source is not available on the episode, can you think of how to hadle that ?